src/rviz/frame_manager.cpp

```diff
@@ -77,10 +77,12 @@ void FrameManager::update()
     cache_.clear();
     switch (sync_mode_)
     {
-    case SyncOff:
+    case SyncOff:   // always use latest time
+    case SyncFrame: // sync to current time
```

Comment:
The problem here is when the TF messages have a small latency w.r.t. ros::Time::now(), which is usually the case in real world scenarios, I think. Then it is likely that one gets continuously extrapolation exceptions in lookupTransform. I think, `sync_time_` should be set to the stamp of the latest available TF. Unfortunately, there are many TFs with different latest stamps.
One way to mitigate this problem would be to check whether it is possible to transform with ros::Time::now() (e.g. with canTransform or lookupTransform + try-catch) in `transform()`. If there is a extrapolation exception fall back to ros::Time(). However, this causes overhead because of additional lookups. And it does not handle the case where a TF message arrives, which is still before ros::Time::now(), during rviz's update cycle.

Reply:
I see your point. I mitigated that problem by substracting a `sync_delta_` from `ros::Time::now()`, which is adapted when a frame lookup fails with an extrapolation exception.